src/test/java/com/wikia/webdriver/common/templates/NewTestTemplate.java

```diff
@@ -49,15 +49,17 @@ public void start(Method method, Object[] data) {
       NewDriverProvider.setUnstablePageLoadStrategy(true);
     }
 
-    String onDriver = method.getAnnotation(Execute.class).allowedDriver();
-    if (onDriver.length() > 0 & !onDriver.equalsIgnoreCase(Configuration.getBrowser())) {
-      throw new SkipException(
-          "The test can not be run on driver " + Configuration
-              .getBrowser() + ". The test is restricted to driver " + onDriver);
-    } else {
-      startBrowser();
+    if (method.isAnnotationPresent(Execute.class)) {
+      String onDriver = method.getAnnotation(Execute.class).allowedDriver();
+      if (onDriver.length() > 0 & !onDriver.equalsIgnoreCase(Configuration.getBrowser())) {
+        throw new SkipException(
+            "The test can not be run on driver " + Configuration
+                .getBrowser() + ". The test is restricted to driver " + onDriver);
+      }
     }
 
+    startBrowser();
+
     if (method.isAnnotationPresent(Execute.class)) {
```

Comment:
@bognix thanks! 
We don't need start a browser if we run the test with inappropriate driver and we shouldn't load a first page if browser is not started. I don't see the way how to combine it, u?